remote-api/src/test/java/org/alfresco/rest/api/impl/rules/ActionParameterConverterTest.java

```diff
@@ -0,0 +1,164 @@
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.impl.rules;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.times;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.alfresco.repo.action.executer.AddFeaturesActionExecuter;
+import org.alfresco.repo.action.executer.CopyActionExecuter;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.action.ActionDefinition;
+import org.alfresco.service.cmr.action.ActionService;
+import org.alfresco.service.cmr.action.ParameterDefinition;
+import org.alfresco.service.cmr.dictionary.DataTypeDefinition;
+import org.alfresco.service.cmr.dictionary.DictionaryService;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@Experimental
+@RunWith(MockitoJUnitRunner.class)
+public class ActionParameterConverterTest
+{
+    private static final String NAMESPACE_URI = "http://www.alfresco.org/model/dictionary/1.0";
+    private static final String VERSIONABLE = "versionable";
+    private static final String CM_PREFIX = "cm";
+    private static final String COLON = ":";
+    private static final String VERSIONABLE_PREFIX = CM_PREFIX + COLON + VERSIONABLE;
+    private static final StoreRef STORE_REF = StoreRef.STORE_REF_WORKSPACE_SPACESSTORE;
+    private static final String DUMMY_FOLDER_NODE_ID = "dummy-folder-node";
+    private static final String SLASH = "/";
+    private static final String DUMMY_FOLDER_NODE_REF = STORE_REF + SLASH + DUMMY_FOLDER_NODE_ID;
+
+    @Mock
+    private DictionaryService dictionaryService;
+    @Mock
+    private ActionService actionService;
+    @Mock
+    private NamespaceService namespaceService;
+
+    @Mock
+    private ActionDefinition actionDefinition;
+    @Mock
+    private ParameterDefinition actionDefinitionParam;
+    @Mock
+    private DataTypeDefinition dataTypeDefinition;
+
+    @InjectMocks
+    private ActionParameterConverter objectUnderTest;
+
+    @Test
+    public void testAddAspectConversion()
+    {
+        final String name = AddFeaturesActionExecuter.NAME;
+        final String aspectNameKey = AddFeaturesActionExecuter.PARAM_ASPECT_NAME;
+        final Map<String, Serializable> params = new HashMap<>(1);
+        params.put(aspectNameKey, VERSIONABLE_PREFIX);
```

Comment:
Assuming we don't need this to be modifiable then we could use:
```
Map.of(aspectNameKey, VERSIONABLE);
```

Reply:
Action parameters are actually modified (each entry is replaced with converted parameter value).
I am thinking about refactoring that somehow.

Reply:
That part is refactored - immutable map can be and is used here.